src/utils/helpers/immutable/immutable.js

```diff
@@ -30,7 +30,7 @@ const ImmutableHelper = {
       return Immutable.Seq(js).map(ImmutableHelper.parseJSON).toList();
     }
 
-    return Immutable.Seq(js).map(ImmutableHelper.parseJSON).toMap();
+    return Immutable.Map(js).map(ImmutableHelper.parseJSON);
```

Comment:
I've been trying to understand why we used `Seq` in the first place, I think the code comes from this example https://github.com/facebook/immutable-js/wiki/Converting-from-JS-objects#custom-conversion

There's also an issue discussing the `length` problem in their GitHub repo https://github.com/facebook/immutable-js/issues/1026

I think `Seq` is used as a little performance boost as it iterates over the nested data with a simpler data object than map. Maybe we could use the advice in the issue linked above and use `Seq.Keyed` to maintain the performance boost?